src/odemis/acq/stream/_sync.py

```diff
@@ -1787,7 +1786,222 @@ def _assembleSpecData(self, data_list, repetition):
+            self._emitter.dwellTime.value = dwell_time
+    
+    def _acquireImage(self, x, y):
+        try: 
+            for ce in self._acq_complete:
+                ce.clear()
+            
+            self._emitter.translation.value = (x, y)
+            # checks the hardware has accepted it
+            trans = self._emitter.translation.value
+            if math.hypot(x - trans[0], y - trans[1]) > 1e-3:
+                logging.warning("Ebeam translation is %s instead of requested %s.", trans, (x, y))
+            
+            # Get data
+            for s, sub in zip(self._streams, self._subscribers):
+                s._dataflow.subscribe(sub)
+            
+            # Wait for detector to acquire image
+            for i, s in enumerate(self._streams):
+                timeout = 1.5 * self._tc_stream._getDetectorVA("dwellTime").value + 1
+                if not self._acq_complete[i].wait(timeout):
+                    raise TimeoutError()
+            if self._acq_state == CANCELLED:
+                raise CancelledError()
+            tc_data, se_data = self._acq_data[-1][-1], self._acq_data[0][-1]
+            return tc_data, se_data
+        finally:
+            for s, sub in zip(self._streams, self._subscribers):
+                s._dataflow.unsubscribe(sub)   
+    
+    def _onCompletedData(self, n, raw_das):        
+        md = raw_das[0].metadata.copy()
+        md[MD_DESCRIPTION] = self._streams[n].name.value
+        md[model.MD_PIXEL_SIZE] = (self._tc_stream.pixelSize.value, self._tc_stream.pixelSize.value)
+        md[model.MD_DWELL_TIME] = self._tc_stream._getDetectorVA("dwellTime").value
+        md[model.MD_DIMS] = "CTZYX"
+        # For now, the viewport cannot display large datasets, so we have to crop the temporal data
+        # FIXME: remove once we can display more data
+        if n == 1:
+            for i, da in enumerate(raw_das):
+                raw_das[i] = da[:, :1024]
+
+        shape = numpy.array(raw_das).shape
+        rep = self._tc_stream.repetition.value
+
+        # The SEM data is of shape 1,1 (YX), while the time-correlator data is of shape 1, 65535 (XT).
+        # So the first dimension can always be discarded and the second dimension can be considered T.
+        # This will not work anymore if we include fuzzing.
+        das = numpy.reshape(raw_das, [shape[2], 1, 1, rep[1], rep[0]])
+        das = model.DataArray(das, md)
+        
+        pos, _ = self._get_center_pxs(self._tc_stream.repetition.value, (1, 1), das[0])
+        das.metadata[model.MD_POS] = pos
+        
+        self._raw.append(das)
+        return das
+
+    def _getNumDriftCors(self):
+        """
+        Returns the number of drift corrections per pixel
+        """
+        if not self._dc_estimator:
+            return 0
+        dc_period = self._dc_estimator.period.value
+        tc_dwell_time = self._tc_stream._getDetectorVA("dwellTime").value
+        return int(tc_dwell_time / dc_period)
```

Comment:
The problem is that now we completely disable the drift correction as soon as the period is smaller than the dwell time. Ideally, it'd fallback to the normal drift correction behaviour (ie, every N pixels) in such case. At least, I'd suggest to only round to 0 if it's below 0.1, and otherwise round it up to1.